src/main/java/mate/academy/intro/controller/OrderController.java

```diff
@@ -0,0 +1,114 @@
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Order Management", description = "Endpoints for managing orders")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+
+    private final OrderService orderService;
+
+    @Operation(summary = "Place a new order")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "201", description = "Order placed successfully",
+                    content = {@Content(mediaType = "application/json",
+                            schema = @Schema(implementation = OrderResponseDto.class))}),
+            @ApiResponse(responseCode = "400",
+                    description = "Invalid input or empty cart", content = @Content)
+    })
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    @PreAuthorize("hasAuthority('ROLE_USER')")
+    public OrderResponseDto placeOrder(@RequestParam Long userId,
+                                       @RequestBody @Valid CreateOrderRequestDto requestDto) {
+        return orderService.placeOrder(userId, requestDto);
+    }
+
+    @Operation(summary = "Get user's orders with pagination")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Found orders",
+                    content = {@Content(mediaType = "application/json",
+                            schema = @Schema(implementation = Page.class))})
+    })
+    @GetMapping
+    @PreAuthorize("hasAuthority('ROLE_USER')")
+    public Page<OrderResponseDto> getOrders(@RequestParam Long userId,
+                                            @PageableDefault(size = 10,
+                                                    sort = "orderDate") Pageable pageable) {
+        return orderService.getUserOrders(userId, pageable);
+    }
+
+    @Operation(summary = "Get all items in an order")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Found order items",
+                    content = {@Content(mediaType = "application/json",
+                            schema = @Schema(implementation = OrderItemResponseDto.class))}),
+            @ApiResponse(responseCode = "404",
+                    description = "Order not found or access denied", content = @Content)
+    })
+    @GetMapping("/{orderId}/items")
+    @PreAuthorize("hasAuthority('ROLE_USER')")
+    public List<OrderItemResponseDto> getOrderItems(@RequestParam Long userId,

```

Comment:
provide Authentication
```suggestion
    public List<OrderItemResponseDto> getOrderItems(
```